python/src/trezorlib/nostr.py

```diff
@@ -0,0 +1,62 @@
+# This file is part of the Trezor project.
+#
+# Copyright (C) 2012-2025 SatoshiLabs and contributors
+#
+# This library is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# as published by the Free Software Foundation.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the License along with this library.
+# If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.
+
+
+import json
+from typing import TYPE_CHECKING, AnyStr
+
+from . import messages
+from .tools import expect
+
+if TYPE_CHECKING:
+    from .client import TrezorClient
+    from .protobuf import MessageType
+    from .tools import Address
+
+
+@expect(messages.NostrPubkey)
+def get_pubkey(
```

Comment:
this function should return just `bytes` of the public key.

perhaps it's a good idea to rebase on #4490

Reply:
Actually I don't think it should be `bytes` but rather the hex-encoded bytes, which is human readable and the (most) standard way to encode Nostr keys. Another encoding commonly used is *NPUB* - and I should probably add an option to return that encoding if desired, but at the most fundamental level it is the hex encoding, which is what all code dealing with Nostr is using. `bytes` seems strange, because it looks bad, can't even be copy-pasted, and is of no use...

Reply:
> `bytes` seems strange, because it looks bad, can't even be copy-pasted, and is of no use...

I mean you're deep inside a library, nobody is supposed to be copy-pasting the return value of this. 
If there is a canonical representation of the pubkey as readable string, the `NostrPubkey` message should send _that_ from Trezor. If there isn't (you apparently have two choices, npub / hex), and you're returning raw bytes, the library function should return raw bytes.

Consumers of `trezorlib.nostr.get_pubkey` are expected to do their own human readable conversion if they need to; specifically, if encoding to npub is simple, feel free to implement that in `cli/nostr.py` and add an option to the user-exposed function

Reply:
(fwiw if you return bytes from a click-exposed function, there's some sort of conversion machinery that will automagically print it as hex for the user)

Reply:
okay, if that's the case, please modify the Trezor side to return the hex string

Reply:
Good point. It's all `str` now: a95dc04d5f42c357a5d00b171c1d780782b55f62 !

Reply:
Back to `bytes` at the protocol and library level, and `str` conversion only in the cli. https://github.com/trezor/trezor-firmware/pull/4436/commits/04e42c873f7c1d900027e37e268dc4df42f72e38